Update to OSX Laptop Dependency Instructions - Patch 1

wretrop opened the pull request:
Added and fleshed out some of the set up instructions for setting up an OSX machine.

Specific details in commit notes.

Also - Let me know if I'm doing this right - It's my first time 😀

wretrop committed:
Update MacOS Instructions in setup_dependencies.md

- Added note to restart comp after installing XQuartz to avoid problems.
- Cleaned up shell config note and added instructions for checking which shell is being used.

wretrop committed:
Update MacOs Set Up Instructions (x2) setup_dependencies.md

- Tweaked shell config instructions to include source reminder
- Added explicit pip3 step since it seems left off
- Eliminated pip3 checkpoint since it becomes redundant with step.

wretrop committed:
Minor Typo -  setup_dependencies.md

- Fixed a typo

wretrop committed:
Updated Instructions for modifying Docker Memory for Mac

Updated the instructions for adjusting memory limit for Docker Desktop on Mac, reflecting current Docker UI.

wretrop committed:
Added troubleshooting for fail auth when running docker hello-world

There appears to be an intermittent bug with Docker Hub regarding connection authorizations, which caused me to fail Hello-World.

I added a troubleshooting step to force re-login.

tanij commented:
@wretrop thanks for the contribution, much appreciated! 

There is a build-breaking mistake in the pull request. (You can't embed a ` ```{} ``` ` environment inside another ` ```{} ``` ` one. You must scale down the ` ``` ` by one. ` ``{} `` ` inside a ` ```{} ``` ` will work.)

For future contributions, always round `dts docs build` locally after making changes, and make sure the build succeeds. You will get a bunch of red WARNINGS - but worry not of those, as long as you get a green "HTML artifacts: [some local address]" box at the end. 

To your credit: your first contribution was in the most intricated part of the docs - no easy feat to get that part right!

tanij committed:
Update setup_docker.md

- Putting new `trouble` inside macOS tab
- Scaling hierarchy of bash environment by one inside the trouble environment

tanij committed:
Update setup_dependencies.md

Minor wording and formatting changes

tanij reviewed (approved):
Commited a few changes.

wretrop commented:
> @wretrop thanks for the contribution, much appreciated!
> 
> There is a build-breaking mistake in the pull request. (You can't embed a `` ```{} ``` `` environment inside another `` ```{} ``` `` one. You must scale down the `` ``` `` by one. `` ``{} `` `` inside a `` ```{} ``` `` will work.)
> 
> For future contributions, always round `dts docs build` locally after making changes, and make sure the build succeeds. You will get a bunch of red WARNINGS - but worry not of those, as long as you get a green "HTML artifacts: [some local address]" box at the end.
> 
> To your credit: your first contribution was in the most intricated part of the docs - no easy feat to get that part right!

Ah, understood - I made my edits "the simple way" (via editor in browser) and couldn't find a way to successfully preview them.  Currently in the process of setting up locally to limit future mistakes!
